R_program/Main.R

```diff
@@ -574,12 +577,91 @@ remove(temp)
 
 ############################# End MARKOV Weather Generator Code ##############################
 
+############################# Phenology Code ###############################
+# This code determines plant functional type phenology
+# and then scales phenological activity, biomass, litter and % live accordingly
+if(rescale_phenology){
+  source(vegetation.file)
+  setwd(db_loc)
+  # Read the input CSV files
+  phenology <- read.csv("InputData_Phenology.csv", header = TRUE, row.names = 1)
+  biomass <- read.csv("InputData_Biomass.csv", header = TRUE, row.names = 1)
+  pctlive <- read.csv("InputData_PctLive.csv", header = TRUE, row.names = 1)
+  litter <- read.csv("InputData_Litter.csv", header = TRUE, row.names = 1)
+  seasons <- read.csv("InputData_GrowingSeason.csv", header = TRUE, row.names = 1)
+  
+  # Turn seasons into a vector of months where plants are expected to grow. 
+  seasons <- Position(function(x) x, seasons):Position(function(x) x, seasons, right = TRUE)
+  
+  # scale_phenology() assumes the months are rows, so we have to transpose.
+  phenology <- t(phenology)
+  pctlive <- t(pctlive)
+  litter <- t(litter)
+  biomass <- t(biomass)
+  
+  # condense the values we want to scale into a single list
+  values_to_scale <- list(phenology, pctlive, litter, biomass)
+  # scale the list
+  scaled_values <- scale_phenology(values_to_scale, sw_weatherList, seasons)
+  
```

Comment:
@chaukap here we do not pass in site_latitude despite it being one of the arguments of scale_phenology. It seems to me we ought to allow the user to set site_latitude here.

Reply:
Parameter added with commit 242771a. The default is site_latitude = 90